discovery/gossiper.go

```diff
@@ -1292,10 +1303,13 @@ func (d *AuthenticatedGossiper) processChanPolicyUpdate(
 			edge: edge,
 		})
 
+		// Finally, add the edge to our policy mapping.
+		chanPolicies[info.ChannelPoint] = edge
```

Comment:
We should only be setting these for channels that have actually been updated (further below).